# Project 2 - Part 1

# An Exploration of 90s Classics

Ruhama Baruda - @02966062

## Chosen Dataset

This dataset features over 900 classic hits from the 1990s, with more than 500 different artists. Each track is described by in-depth Spotify audio features data such as danceability, energy, key, loudness, mode, speechiness, acousticness, instrumentalness, liveness, valence, tempo, and duration. It provides a wide range of variables to give insight into the musical landscape of the 90s, allowing for an exploration of what made music in that era popular and distinct as well as relationships between musical features.

### Source Description

The dataset chosen for this project is titled "1990s Classic Hits (with Spotify Data)" and was sourced from Kaggle. The dataset was uploaded to Kaggle by retired computer programmer and music fanatic, The Bumpkin, in August 2024

Link to Kaggle post: https://www.kaggle.com/datasets/thebumpkin/1990s-classic-hits-with-spotify-data

## Import Pandas and Data Set

In [2]:
import pandas as pd
df = pd.read_csv('1990sClassicHits.csv') #import csv file
df.head() #first few rows of dataset

Track          Artist Duration  Time_Signature  Danceability  \
0  Everybody Everybody       Black Box     4:11               4         0.645   
1         Close To You     Maxi Priest     3:45               4         0.590   
2               Do Me!  Bell Biv DeVoe     3:56               4         0.615   
3           Everything     Jody Watley     4:05               4         0.704   
4        What It Takes       Aerosmith     3:21               4         0.672   

   Energy  Key  Loudness  Mode  Speechiness  Acousticness  Instrumentalness  \
0   0.795    7   -12.095     1       0.0527        0.3470          0.003890   
1   0.727    6    -7.955     1       0.0706        0.0259          0.000065   
2   0.534    9    -6.719     0       0.0386        0.1060          0.000018   
3   0.225    6   -14.454     0       0.0994        0.9020          0.657000   
4   0.655   10    -5.021     0       0.0310        0.0362          0.000000   

   Liveness  Valence    Tempo  Popularity  Year  
0    0.1040    0.535  112.067          87  1990  
1    0.0827    0.356  121.966          85  1990  
2    0.0607    0.193  135.917          84  1990  
3    0.1060    0.243  120.006          82  1990  
4    0.1170    0.555  134.948          82  1990

## Section 1: Dataset Overview and Structure

### 1. What is the shape and structure of the dataset?

In [3]:
print("The first 5 rows of the table:\n")
df.head(5) #first 5 rows of the data set

The first 5 rows of the table:



Track          Artist Duration  Time_Signature  Danceability  \
0  Everybody Everybody       Black Box     4:11               4         0.645   
1         Close To You     Maxi Priest     3:45               4         0.590   
2               Do Me!  Bell Biv DeVoe     3:56               4         0.615   
3           Everything     Jody Watley     4:05               4         0.704   
4        What It Takes       Aerosmith     3:21               4         0.672   

   Energy  Key  Loudness  Mode  Speechiness  Acousticness  Instrumentalness  \
0   0.795    7   -12.095     1       0.0527        0.3470          0.003890   
1   0.727    6    -7.955     1       0.0706        0.0259          0.000065   
2   0.534    9    -6.719     0       0.0386        0.1060          0.000018   
3   0.225    6   -14.454     0       0.0994        0.9020          0.657000   
4   0.655   10    -5.021     0       0.0310        0.0362          0.000000   

   Liveness  Valence    Tempo  Popularity  Year  
0    0.1040    0.535  112.067          87  1990  
1    0.0827    0.356  121.966          85  1990  
2    0.0607    0.193  135.917          84  1990  
3    0.1060    0.243  120.006          82  1990  
4    0.1170    0.555  134.948          82  1990

In [4]:
print("The last 5 rows of the table:\n")
df.tail(5) #last 5 rows of the data set

The last 5 rows of the table:



Track                                Artist Duration  \
914           I’m Your Angel              R. Kelly and Celine Dion     3:12   
915        Taking Everything                         Gerald Levert     5:25   
916     What’s It Gonna Be?!          Busta Rhymes featuring Janet     2:41   
917     Almost Doesn’t Count                                Brandy     3:38   
918  It Ain’t My Fault 1 & 2  Silkk the Shocker Featuring Mystikal     3:30   

     Time_Signature  Danceability  Energy  Key  Loudness  Mode  Speechiness  \
914               4         0.648   0.331    4   -10.952     1       0.0274   
915               4         0.689   0.710    3    -5.395     0       0.0413   
916               4         0.710   0.742    9    -4.699     0       0.0339   
917               4         0.682   0.543    0    -7.063     0       0.0454   
918               4         0.743   0.567    4   -12.468     0       0.2570   

     Acousticness  Instrumentalness  Liveness  Valence    Tempo  Popularity  \
914        0.7760          0.000445     0.126    0.507   98.572          28   
915        0.0886          0.000000     0.107    0.306  113.722          26   
916        0.1780          0.000064     0.080    0.504  127.969          26   
917        0.4940          0.000000     0.158    0.478   71.582          19   
918        0.1960          0.000000     0.228    0.755   90.096           0   

     Year  
914  1999  
915  1999  
916  1999  
917  1999  
918  1999

In [5]:
# General information of the DataFrame
print("The general information of the data:\n")
df.info()
print("The number of rows and columns of the data:\n")
df.shape

The general information of the data:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 919 entries, 0 to 918
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Track             919 non-null    object 
 1   Artist            919 non-null    object 
 2   Duration          919 non-null    object 
 3   Time_Signature    919 non-null    int64  
 4   Danceability      919 non-null    float64
 5   Energy            919 non-null    float64
 6   Key               919 non-null    int64  
 7   Loudness          919 non-null    float64
 8   Mode              919 non-null    int64  
 9   Speechiness       919 non-null    float64
 10  Acousticness      919 non-null    float64
 11  Instrumentalness  919 non-null    float64
 12  Liveness          919 non-null    float64
 13  Valence           919 non-null    float64
 14  Tempo             919 non-null    float64
 15  Popularity        919 non-null    int64  
 16  Year  

(919, 17)

### 2. Identify the variable types

In [6]:
#display data type of each column
print("The data type of each column:\n")
df.dtypes

The data type of each column:



Track                object
Artist               object
Duration             object
Time_Signature        int64
Danceability        float64
Energy              float64
Key                   int64
Loudness            float64
Mode                  int64
Speechiness         float64
Acousticness        float64
Instrumentalness    float64
Liveness            float64
Valence             float64
Tempo               float64
Popularity            int64
Year                  int64
dtype: object

#### Selecting columns by data type

In [8]:
print("Numerical columns:\n")
df.select_dtypes(include=['number']) #numerical data types

Numerical columns:



Time_Signature  Danceability  Energy  Key  Loudness  Mode  Speechiness  \
0                 4         0.645   0.795    7   -12.095     1       0.0527   
1                 4         0.590   0.727    6    -7.955     1       0.0706   
2                 4         0.615   0.534    9    -6.719     0       0.0386   
3                 4         0.704   0.225    6   -14.454     0       0.0994   
4                 4         0.672   0.655   10    -5.021     0       0.0310   
..              ...           ...     ...  ...       ...   ...          ...   
914               4         0.648   0.331    4   -10.952     1       0.0274   
915               4         0.689   0.710    3    -5.395     0       0.0413   
916               4         0.710   0.742    9    -4.699     0       0.0339   
917               4         0.682   0.543    0    -7.063     0       0.0454   
918               4         0.743   0.567    4   -12.468     0       0.2570   

     Acousticness  Instrumentalness  Liveness  Valence    Tempo  Popularity  \
0          0.3470          0.003890    0.1040    0.535  112.067          87   
1          0.0259          0.000065    0.0827    0.356  121.966          85   
2          0.1060          0.000018    0.0607    0.193  135.917          84   
3          0.9020          0.657000    0.1060    0.243  120.006          82   
4          0.0362          0.000000    0.1170    0.555  134.948          82   
..            ...               ...       ...      ...      ...         ...   
914        0.7760          0.000445    0.1260    0.507   98.572          28   
915        0.0886          0.000000    0.1070    0.306  113.722          26   
916        0.1780          0.000064    0.0800    0.504  127.969          26   
917        0.4940          0.000000    0.1580    0.478   71.582          19   
918        0.1960          0.000000    0.2280    0.755   90.096           0   

     Year  
0    1990  
1    1990  
2    1990  
3    1990  
4    1990  
..    ...  
914  1999  
915  1999  
916  1999  
917  1999  
918  1999  

[919 rows x 14 columns]

In [9]:
#Time_Signature and Mode should be categorical
#converting numerical columns to categorical (Time_Signature, Mode, and Year)
df["Time_Signature"] = df["Time_Signature"].astype('category')
df["Mode"] = df["Mode"].astype('category')

In [10]:
print("Categorical columns:\n")
df.select_dtypes(include=['object', 'category'])

Categorical columns:



Track                                Artist Duration  \
0        Everybody Everybody                             Black Box     4:11   
1               Close To You                           Maxi Priest     3:45   
2                     Do Me!                        Bell Biv DeVoe     3:56   
3                 Everything                           Jody Watley     4:05   
4              What It Takes                             Aerosmith     3:21   
..                       ...                                   ...      ...   
914           I’m Your Angel              R. Kelly and Celine Dion     3:12   
915        Taking Everything                         Gerald Levert     5:25   
916     What’s It Gonna Be?!          Busta Rhymes featuring Janet     2:41   
917     Almost Doesn’t Count                                Brandy     3:38   
918  It Ain’t My Fault 1 & 2  Silkk the Shocker Featuring Mystikal     3:30   

    Time_Signature Mode  
0                4    1  
1                4    1  
2                4    0  
3                4    0  
4                4    0  
..             ...  ...  
914              4    1  
915              4    0  
916              4    0  
917              4    0  
918              4    0  

[919 rows x 5 columns]

### 3. Are there any missing values?

In [11]:
#identifying missing values
print("Number of missing values by column:\n")
df.isnull().sum()

Number of missing values by column:



Track               0
Artist              0
Duration            0
Time_Signature      0
Danceability        0
Energy              0
Key                 0
Loudness            0
Mode                0
Speechiness         0
Acousticness        0
Instrumentalness    0
Liveness            0
Valence             0
Tempo               0
Popularity          0
Year                0
dtype: int64

In [12]:
print("There are no missing values.")

There are no missing values.


## Section 2: Data Cleaning & Indexing

### 4. Check for and remove duplicates

In [13]:
#checking for duplicated rows
duplicates_num = df.duplicated().sum()
print(f"Number of duplicate rows:\n {duplicates_num}")

Number of duplicate rows:
 0


### 5. Provide two examples of DataFrame filtering using logical conditions

#### Filtering by column labels

In [14]:
#selecting columns: Track, Artist, Popularity, and Year
print("Filtered DataFrame showing tracks, artists, popularity, and year:\n")
df.loc[:, ["Track", "Artist", "Popularity", "Year"]]

Filtered DataFrame showing tracks, artists, popularity, and year:



Track                                Artist  \
0        Everybody Everybody                             Black Box   
1               Close To You                           Maxi Priest   
2                     Do Me!                        Bell Biv DeVoe   
3                 Everything                           Jody Watley   
4              What It Takes                             Aerosmith   
..                       ...                                   ...   
914           I’m Your Angel              R. Kelly and Celine Dion   
915        Taking Everything                         Gerald Levert   
916     What’s It Gonna Be?!          Busta Rhymes featuring Janet   
917     Almost Doesn’t Count                                Brandy   
918  It Ain’t My Fault 1 & 2  Silkk the Shocker Featuring Mystikal   

     Popularity  Year  
0            87  1990  
1            85  1990  
2            84  1990  
3            82  1990  
4            82  1990  
..          ...   ...  
914          28  1999  
915          26  1999  
916          26  1999  
917          19  1999  
918           0  1999  

[919 rows x 4 columns]

#### Filtering by rows

In [15]:
#selecting rows where year is after 1995
print("Songs released after 1995:\n")
yearfilter_df = df.loc[:, ["Track", "Artist", "Year"]]
yearfilter_df[yearfilter_df.Year > 1995]

Songs released after 1995:



Track                                Artist  Year
565                   Nobody     Keith Sweat featuring Athena Cage  1996
566                     Time                 Hootie & the Blowfish  1996
567                      Hay                      Crucial Conflict  1996
568                 Jealousy                      Natalie Merchant  1996
569              Be My Lover                             La Bouche  1996
..                       ...                                   ...   ...
914           I’m Your Angel              R. Kelly and Celine Dion  1999
915        Taking Everything                         Gerald Levert  1999
916     What’s It Gonna Be?!          Busta Rhymes featuring Janet  1999
917     Almost Doesn’t Count                                Brandy  1999
918  It Ain’t My Fault 1 & 2  Silkk the Shocker Featuring Mystikal  1999

[354 rows x 3 columns]

In [16]:
#selecting rows where danceability is above 0.5 (scale of 0-1.0)
print("Songs that score above 0.5 in danceability:\n")
#new df with relevant columns
dance_df = df.loc[:, ["Track", "Artist", "Danceability"]]
#select rows where danceability value is greater than 0.5
dance_dfsort = dance_df[dance_df.Danceability > 0.5]
#sort new df in ascending order
dance_dfsort.sort_values(by = "Danceability")

Songs that score above 0.5 in danceability:



Track                     Artist  Danceability
153     P.A.S.S.I.O.N.            Rythm Syndicate         0.501
643            Someone         SWV and Puff Daddy         0.501
479    Someone To Love  Jon B. featuring Babyface         0.501
363    I’m So Into You                        SWV         0.501
730     It’s Your Love  Tim McGraw and Faith Hill         0.502
..                 ...                        ...           ...
353  Three Little Pigs                Green Jellÿ         0.930
383        Shoop (#5*)               Salt N’ Pepa         0.938
408              Shoop                Salt-n-Pepa         0.939
786           Westside                         TQ         0.949
28        Ice Ice Baby                Vanilla Ice         0.979

[717 rows x 3 columns]

## Section 3: Descriptive Statistics

### 6. Use .describe() on numerical columns

In [17]:
#new df with numerical columns only
num_columns = df.select_dtypes(include=['number'])
#descriptive stats of new df
num_columns.describe()

Danceability      Energy         Key    Loudness  Speechiness  \
count    919.000000  919.000000  919.000000  919.000000   919.000000   
mean       0.622967    0.623548    5.412405   -8.349071     0.075793   
std        0.154782    0.203975    3.514783    3.892821     0.079570   
min        0.135000    0.001340    0.000000  -42.950000     0.023000   
25%        0.518500    0.479500    2.000000  -10.166500     0.032400   
50%        0.631000    0.639000    6.000000   -7.599000     0.042900   
75%        0.738000    0.780000    9.000000   -5.751000     0.075600   
max        0.979000    0.999000   11.000000    0.919000     0.529000   

       Acousticness  Instrumentalness    Liveness     Valence       Tempo  \
count    919.000000        919.000000  919.000000  919.000000  919.000000   
mean       0.230068          0.052947    0.171865    0.531424  118.711789   
std        0.256284          0.180214    0.138296    0.246091   29.684070   
min        0.000004          0.000000    0.025000    0.006720   48.718000   
25%        0.029700          0.000000    0.089200    0.341500   96.924000   
50%        0.120000          0.000006    0.118000    0.535000  115.010000   
75%        0.344500          0.000741    0.217500    0.728500  133.424000   
max        0.995000          0.987000    0.965000    0.975000  219.973000   

       Popularity         Year  
count  919.000000   919.000000  
mean    57.722524  1994.405876  
std     17.599522     2.926405  
min      0.000000  1990.000000  
25%     49.000000  1992.000000  
50%     60.000000  1994.000000  
75%     70.000000  1997.000000  
max     92.000000  1999.000000

Insights:
- On average,1990s classic hits were danceable and energetic.
- The average tempo of the music in this dataset 118.7

### 7. Use .value_counts() to analyze at least two categorical columns

In [18]:
#new df with categorical columns only
cat_columns = df.select_dtypes(include=['category', 'object'])
#value counts for artist column only
cat_columns["Artist"].value_counts()

Artist
Mariah Carey                            16
Madonna                                 14
Janet Jackson                           13
Boyz II Men                             11
Whitney Houston                         10
                                        ..
Duice                                    1
4 Non Blondes                            1
Tina Turner                              1
The Heights                              1
Silkk the Shocker Featuring Mystikal     1
Name: count, Length: 535, dtype: int64

In [22]:
#convert year to categorical data to count how many tracks each year
cat_year_df = df["Year"].astype('category')
#count how many tracks in each year
print("Count of tracks in each year:\n")
cat_year_df.value_counts()

Count of tracks in each year:



Year
1991    100
1990     99
1999     96
1992     95
1993     94
1994     91
1998     91
1997     89
1995     86
1996     78
Name: count, dtype: int64

In [21]:
#value counts for track column only
print("Count of track titles:\n")
cat_columns["Track"].value_counts()

Count of track titles:



Track
Hold On                         3
I’m The Only One                2
I Don’t Want To Miss A Thing    2
Stay                            2
Always                          2
                               ..
Bed Of Roses                    1
Insane In The Brain             1
Layla                           1
Baby I’m Yours                  1
It Ain’t My Fault 1 & 2         1
Name: count, Length: 892, dtype: int64

### 8. Use .groupby() to compute summary statistics across categories

In [23]:
#new df with relevant columns
group_df = df.loc[:, ["Track", "Artist", "Popularity", "Danceability"]]

# group by Artist and find averages of popularity and danceability
artist_grouped = group_df.groupby("Artist")[["Popularity", "Danceability"]].mean()
print("Average popularity and danceability of each artist:\n")
artist_grouped

Average popularity and danceability of each artist:



Popularity  Danceability
Artist                                                               
10,000 Maniacs                                     66.0         0.466
112                                                76.0         0.720
112 featuring Lil Zane                             57.0         0.475
112 featuring The Notorious B.I.G. and Mase        69.0         0.490
20 Fingers featuring Gillette                      47.0         0.910
...                                                 ...           ...
by Method Man featuring Mary J. Blige               0.0         0.629
the Beatles                                        48.0         0.474
the Righteous Brothers                             77.0         0.339
‘N Sync                                            64.0         0.422
‘N Sync and Gloria Estefan                         39.0         0.375

[535 rows x 2 columns]

## Section 4: Observations & Questions

### 9. Highlight 1–2 key takeaways or patterns from your descriptive analysis

One major takeaway from this dataset is that 90s hit music tend to be more danceable and energetic.

Another key pattern in the dataset is that the three most featured artists in the list of hits are women: Mariah Carey, Madonna, and Janet Jackson.

### 10. Suggest 2 questions you would explore further in Part 2 using visualization

1. What is the distribution of track durations?
2. Is there a correlation between danceability and tempo?  